Formula/clazy.rb

```diff
@@ -70,7 +74,18 @@ def install
     EOS
 
     ENV["CLANGXX"] = Formula["llvm"].opt_bin/"clang++"
-    system "cmake", "-DCMAKE_CXX_COMPILER=#{bin}/clazy", "."
+    flags = ["-DCMAKE_CXX_COMPILER=#{bin}/clazy"]
+    # Add approprite CMake flags on Linux to make sure clazy uses headers and shared libraries
+    # from GCC 11, not host headers and libraries, because the host copies may be too old.
+    unless OS.mac?
+      flags += [
+        "-DCMAKE_CXX_FLAGS=-I#{Formula["gcc"].opt_include}/c++/11
+          -I#{Formula["gcc"].opt_include}/c++/11/x86_64-pc-linux-gnu",
+        "-DCMAKE_EXE_LINKER_FLAGS=-L#{Formula["gcc"].opt_lib}/gcc/11
+          -Wl,-rpath,#{Formula["gcc"].opt_lib}/gcc/11",
+      ]
+    end
```

Comment:
This seems like something that users shouldn't need to set for themselves to use, no?

Reply:
This is actually a really good example of what I was talking about on Slack recently - `clang` has no equivalent to GCC spec files, so there's no way for it to find C++ run time libraries or headers if they are not in the default search paths.  This is true whether we use `libc++` or `libstdc++`. I think we avoid the issue on macOS because we can rely on the host C++ libraries/headers being new enough (and backwards compatible).  On Linux it seems we are in a tougher position because there doesn't seem to be a way to automatically add these flags the way we can with GCC.

Reply:
Hmmm, I wonder if there's a way to wrap `clazy` in a script to get around this.

In any case, since we write a `CMakeLists.txt` in this test, we should probably move more of this configuration there instead of adding command-line flags.

Reply:
Yeah I do wonder if for `clazy`, `clang` and `openmpi` (it also has this problem) we could ship extremely simplified "shims" that just wrap the compiler commands in a script adds the extra flags.

For adding these flags to `CMakelists.txt`, should I just make a different one on Linux and macOS?  I don't think we want to add these flags on macOS and I'm not entirely sure how to only add lines on Linux.

Reply:
You can append the relevant lines conditionally on Linux before writing to the file. CMake also supports `if` statements in `CMakeLists.txt`.

Reply:
Using an `if` statement within `CMakeLists.txt` worked great!  Conveniently CMake seems to automatically add the RPATH and `-L` flags both just with `link_directories`, so this cleans up very nicely though.  In the future I think it would be great to wrap both `clazy` and `clang` (and any other compile which needs GCC libs) in scripts to avoid the need for this, but at least now users have a working example of a `CMakeLists.txt` they could copy for building their own projects.